test/acquisition/multi_objective/test_monte_carlo.py

```diff
@@ -527,6 +527,65 @@ def test_constrained_q_expected_hypervolume_improvement(self):
                 )
                 res = acqf(X)
                 self.assertAlmostEqual(res.item(), 0.5 * 1.5, places=4)
+            # test zero slack multiple constraints
+            for eta in (1e-1, 1e-2):
+                acqf = qExpectedHypervolumeImprovement(
+                    model=mm,
+                    ref_point=ref_point,
+                    partitioning=partitioning,
+                    sampler=sampler,
+                    constraints=[
+                        lambda Z: torch.zeros_like(Z[..., -1]),
+                        lambda Z: torch.zeros_like(Z[..., -1]),
+                    ],
+                    eta=eta,
+                )
+                res = acqf(X)
+                self.assertAlmostEqual(res.item(), 0.5 * 0.5 * 1.5, places=4)
+            # test multiple constraints one eta
+            acqf = qExpectedHypervolumeImprovement(
+                model=mm,
+                ref_point=ref_point,
+                partitioning=partitioning,
+                sampler=sampler,
+                constraints=[
+                    lambda Z: torch.ones_like(Z[..., -1]),
+                    lambda Z: torch.ones_like(Z[..., -1]),
+                ],
+                eta=1.0,
+            )
+            res = acqf(X)
+            self.assertAlmostEqual(
+                res.item(),
+                (
+                    torch.sigmoid(torch.as_tensor(-1.0))
+                    * torch.sigmoid(torch.as_tensor(-1.0))
+                    * 1.5
+                ).item(),
+                places=4,
+            )
```

Comment:
It'd be good to reduce the repetition in these test as well. Looks like only eta and the expected value changes. Maybe you can run them in a for loop?

Reply:
Yeah, I introduced loops for this. In principle one could have it even more concise but then one would loose readability. Personally, I really like the `@pytest.mark.parametrize` decorator for these kind of tests, as they make the code very condense but still readable. But I think you are not using it in botorch.